igvm/commands.py

```diff
@@ -611,20 +611,12 @@ def _get_hypervisor(hostname, ignore_reserved=False):
 
 @contextmanager
 def _get_best_hypervisor(vm, hypervisor_states, offline=False):
-    hypervisors = list()
-    for hv_obj in Query({
+    hypervisors = (Hypervisor(o) for o in Query({
         'servertype': 'hypervisor',
         'environment': environ.get('IGVM_MODE', 'production'),
         'vlan_networks': vm.dataset_obj['route_network'],
         'state': Any(*hypervisor_states),
-    }, HYPERVISOR_ATTRIBUTES):
-        try:
-            hypervisors.append(Hypervisor(hv_obj))
-        except libvirtError:
-            log.warning(
-                'Could not connect to HV {}, skipping it'
-                .format(hv_obj['hostname'])
-            )
```

Comment:
I see why you are removing this here now. Shout out to @kofrezo for seeing this one first.

You still need to catch the libvirtError somewhere though. Maybe in self.conn()?

Reply:
The place where connection might be initiated and fail is `hypervisor.check_vm()` at line 634. This basically return things to previous behaviour, which was correct.

Reply:
Still there might be an libvirt error raised by the connection as @HerrSpace mentioned we better cover it. I think catching it at conn method is a good idea.

Reply:
No, libvirtError is never raised by Hypervisor constructor anymore.

Reply:
Ah, indeed. Cool!